src/librustc_mir/monomorphize/partitioning.rs

```diff
@@ -872,19 +873,28 @@ fn collect_and_partition_mono_items(
 
     tcx.sess.abort_if_errors();
 
-    assert_symbols_are_distinct(tcx, items.iter());
-
-    let strategy = if tcx.sess.opts.incremental.is_some() {
-        PartitioningStrategy::PerModule
-    } else {
-        PartitioningStrategy::FixedUnitCount(tcx.sess.codegen_units())
-    };
-
-    let codegen_units = time(tcx.sess, "codegen unit partitioning", || {
-        partition(tcx, items.iter().cloned(), strategy, &inlining_map)
-            .into_iter()
-            .map(Arc::new)
-            .collect::<Vec<_>>()
+    let (codegen_units, _) = time(tcx.sess, "partition and assert distinct symbols", || {
```

Comment:
Why are we only putting these two in parallel? At least it looks like there's no dependency on codegen units until after we collect `mono_items`, shouldn't that also be a (third) parallel step here?

Reply:
That step is pretty cheap so there isn't much point.

Reply:
It seems like that's not really for us to decide (and rayon should hopefully deal with "pretty cheap" pretty well), but I'm not opposed to leaving it as is.